pkgs/development/python-modules/unsloth/default.nix

```diff
@@ -89,6 +91,19 @@ buildPythonPackage rec {
     torchvision
   ];
 
+  propagatedBuildInputs = [ stdenv.cc.cc ];
+
+  postInstall = ''
+    # Provide a C/C++ compiler to Triton's runtime JIT.
+    # A .pth file executes at interpreter startup and sets env only if not already set.
+    cat > $out/${python.sitePackages}/_unsloth_cc.pth <<'PY'
+import os
+os.environ.setdefault("CC", "${stdenv.cc.cc}/bin/cc")
```

Comment:
Interesting trick, but may lead to surprising behaviour? The environment variable would be set automatically for all python processes if unsloth is merely part (possibly unused) of their closure

Also from my reading of https://docs.python.org/3/library/site.html, only the lines actually beginning with `import` are "executed", so you probably meant to split by `;`?